publication-rest/src/test/java/no/unit/nva/publication/fetch/FetchPublicationHandlerTest.java

```diff
@@ -514,10 +515,42 @@ void shouldReturnUnpublishedPublicationWithFilesWhenUserHasRightsToUpdatedPublic
         var publicationResponse = JsonUtils.dtoObjectMapper.readValue(gatewayResponse.getBody(),
                                                                       PublicationResponseElevatedUser.class);
 
+        var expectedEtag = Resource.fromPublication(publication).fetch(publicationService).orElseThrow().getVersion();
+
+        assertEquals(gatewayResponse.getHeaders().get(ETAG), expectedEtag.toString());
         assertThat(gatewayResponse.getStatusCode(), is(equalTo(HTTP_OK)));
         assertThat(publicationResponse.getAssociatedArtifacts(), is(not(emptyIterable())));
     }
 
+    @Test
+    void shouldReturnPublicationVersionAsETagHeader() throws ApiGatewayException, IOException {
+        var publication = createPublication();
+        fetchPublicationHandler.handleRequest(generateHandlerRequest(publication.getIdentifier().toString()), output,
+                                              context);
+        var gatewayResponse = parseHandlerResponse();
+        var expectedEtag = Resource.fromPublication(publication).fetch(publicationService).orElseThrow().getVersion();
+
+        assertEquals(gatewayResponse.getHeaders().get(ACCESS_CONTROL_EXPOSE_HEADERS), ETAG);
+        assertEquals(String.valueOf(expectedEtag), gatewayResponse.getHeaders().get(ETAG));
+    }
+
+    @Test
+    void shouldReturnNotModifiedWhenProvidingIfNoneMatchHeaderWithETagMatchingCurrentVersionOfPublication()
+        throws ApiGatewayException, IOException {
+        var publication = createPublication();
+        var version = Resource.fromPublication(publication).fetch(publicationService).orElseThrow().getVersion();
```

Comment:
create a test that sends garbage in version? should return latest object probably?